### ISSUE TITLE
Crash in `gs_texture_set_image()` when closing filters dialog

### ISSUE THREAD

**Issue by tt2468** [obsproject/obs-studio#4579]
### Operating System Info

Ubuntu Server 20.04 with xfce4.

nvidia-smi output:
```
Fri Apr 23 09:18:27 2021
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.100      Driver Version: 440.100      CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla T4            Off  | 00000000:00:04.0 Off |                    0 |
| N/A   39C    P8     9W /  70W |    131MiB / 15109MiB |      1%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0       807      G   /usr/lib/xorg/Xorg                           127MiB |
|    0      1101      G   xfwm4                                          3MiB |
+-----------------------------------------------------------------------------+
```

### OBS Studio Version

Git

### OBS Studio Version (Other)

Version up to commit https://github.com/obsproject/obs-studio/commit/244b6c92e66019c5fb07cfab28b6f7a3f35b5e79

### OBS Studio Log URL

https://pastebin.com/w8WYbabE

### OBS Studio Crash Log URL

https://pastebin.com/naBez8GS

### Expected Behavior

The destination pointer should be valid

### Current Behavior

The destination pointer of memcpy is invalid and OBS segfaults

### Steps to Reproduce

1. Open OBS
2. **Disable the main preview**
2. Open filters dialog or source properties
3. Close filters dialog or source properties
4. May take a few times of repeating 3 and 4. Each time it throws a bunch of GLXBadDrawable errors. Occasionally logs `No dest!` and crashes.

I cannot seem to reproduce the issue when the OBS preview is enabled.

### Anything else we should know?

Added
```c
		if (!ptr)
			blog(LOG_INFO, "No dest!");
		if (!data)
			blog(LOG_INFO, "No data!");
```
after line 1168 in libobs/graphics/graphics.c to check for invalid pointers. When bug is reproduced, the destination pointer provided to memcpy is invalid.

I am able to provide remote access to anyone who would like to debug more.

**Comment by tt2468** [comment]
Update:

Tried my fix in #4576, and it still crashes on 440.

Tested upgrading from nvidia-440 to nvidia-450, and it no longer crashes when using the patch in #4576.

**Comment by tt2468** [comment]
Closing. This appears to be the same issue as #4576 fixes
